src/datadoc/backend/utils.py

```diff
@@ -75,9 +65,254 @@ def derive_assessment_from_state(state: DataSetState) -> Assessment:
+    Example:
+        >>> incorrect_date_order(datetime.date(1980, 1, 1), datetime.date(1967, 1, 1))
+        True
+
+        >>> incorrect_date_order(datetime.date(1967, 1, 1), datetime.date(1980, 1, 1))
+        False
+    """
+    return date_from is not None and date_until is not None and date_until < date_from
+
+
+def num_obligatory_dataset_fields_completed(dataset: model.Dataset) -> int:
+    """Count the number of obligatory dataset fields that have values.
+
+    This function returns the total count of obligatory fields in the dataset that are not None.
+
+    Args:
+        dataset (model.Dataset): The dataset object for which to count the fields.
+
+    Returns:
+        int: The number of obligatory dataset fields that have been completed (not None).
+    """
+    return len(
+        [
+            k
+            for k, v in dataset.model_dump().items()
+            if k in OBLIGATORY_DATASET_METADATA_IDENTIFIERS and v is not None
+        ],
+    )
+
+
+def num_obligatory_variables_fields_completed(variables: list) -> int:
+    """Count the number of obligatory fields completed for each variable.
+
+    This function calculates the total number of obligatory fields that have values
+    for each variable in the list.
+
+    Args:
+        variables (list): A list of variable objects to count obligatory fields for.
+
+    Returns:
+        int: The total number of obligatory variable fields that have been completed
+        (not None) across all variables.
+    """
+    num_variables = 0
+    for variable in variables:
+        num_variables = len(
+            [
+                k
+                for k, v in variable.model_dump().items()
+                if k in OBLIGATORY_VARIABLES_METADATA_IDENTIFIERS and v is not None
+            ],
+        )
+    return num_variables
+
+
+def get_missing_obligatory_dataset_fields(dataset: model.Dataset) -> list:
+    """Identify all obligatory dataset fields that are missing values.
+
+    This function checks for obligatory fields that are either directly missing (i.e., set to `None`) or have
+    multilanguage values with empty content.
+
+    Args:
+        dataset (model.Dataset): The dataset object to examine. This object must support the `model_dump()` method which returns a dictionary of field names and values.
+
+    Returns:
+        list: A list of field names (as strings) that are missing values. This includes:
+            - Fields that are directly `None` and are listed in `OBLIGATORY_DATASET_METADATA_IDENTIFIERS`.
+            - Multilanguage fields (identified by `OBLIGATORY_DATASET_METADATA_IDENTIFIERS_MULTILANGUAGE`) where
+                the value exists but the primary language text is empty.
+    """
+    missing_dataset_fields = [
+        k
+        for k, v in dataset.model_dump().items()
+        if k in OBLIGATORY_DATASET_METADATA_IDENTIFIERS and v is None
+    ]
+    missing_multilanguage_fields = [
+        k
+        for k, value in dataset.model_dump().items()
```

Comment:
Would it not be easier to work with these as objects rather than dicts? Could even be possible to implement a method in LanguageStringType to check whether it's empty or not.

Reply:
It could be interesting with a method in LanguageStringType, it was a bit "messy" working with this...

Reply:
But for now I added some helper methods...